src/agni/client/receiver/ChatReceiver.java

```diff
@@ -1,5 +1,32 @@
 package agni.client.receiver;
 
+import agni.client.view.AgniClientView;
+
 public class ChatReceiver extends MessageParser {
 
+    public ChatReceiver() {
+
+    }
+
+    private void notifyChatReceived(String src, String message) {
+
+    }
+
+    @Override
+    public void register(AgniClientView listener) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void deregister(AgniClientView listener) {
```

Comment:
The `register` and `deregister()` methods should not be implemented in the Receivers, they should inherit the implementations from the `MessageParser` since this implementation will be the same throughout all the `Receivers`.